modules/build/src/main/scala/scala/build/CrossSources.scala

```diff
@@ -152,6 +152,41 @@ object CrossSources {
         .map(_.flatten)
 
     val preprocessedInputFromArgs = value(preprocessSources(inputs.flattened()))
+    val preprocessedWithUsingDirs = preprocessedInputFromArgs.filter(_.directivesLocation.isDefined)
+
+    if (preprocessedWithUsingDirs.length > 1) {
+      val projectFilePath = inputs.elements.projectSettingsFiles.headOption match
+        case Some(s) => s.path
+        case _       => inputs.workspace / "project.scala"
+      preprocessedWithUsingDirs
+        .filter(_.scopePath != ScopePath.fromPath(projectFilePath))
+        .foreach { source =>
+          source.directivesLocation match
+            case Some(location) =>
+              val latestLocation = Seq(
+                location.codeDirectives,
+                location.specialCommentDirectives,
+                location.plainCommentDirectives
+              ).maxBy(_.line)
+              val (pathEither, pathStr) = source.scopePath.root match
+                case Right(base: os.Path) =>
+                  val path = os.Path(source.scopePath.path, base)
+                  (Right(path), path.toString)
+                case Left(str: String) =>
+                  val path = str + source.scopePath.path
+                  (Left(path), path)
+              val position = Position.File(
+                pathEither,
+                (0, 0),
+                (latestLocation.line, latestLocation.column)
+              )
+              logger.diagnostic(
+                s"Using directives detected in $pathStr. It is recommended to keep them centralized in the $projectFilePath file.",
```

Comment:
In metals, I don't think it's useful to print the path in the message alert.

And in the command line I think the path is printed twice because by default the warning is printed with the file path if the `position` is set.